src/foam/util/UIDGenerator.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.util',
+  name: 'UIDGenerator',
+  flags: ['java'],
+
+  javaImports: [
+    'java.util.Arrays',
+    'java.util.concurrent.ThreadLocalRandom',
+    'java.util.List'
+  ],
+
+  properties: [
+    {
+      name: 'seqNo',
+      class: 'Int',
+      javaFactory: 'return 0;'
+    },
+    {
+      name: 'lastTimeCalled',
+      class: 'Long',
+      javaFactory: 'return System.currentTimeMillis() / 1000;'
+    },
+    {
+      name: 'randInts',
+      class: 'List',
+      javaType: 'List<Integer>',
+      javaFactory: `
+        List<Integer> randInts = Arrays.asList(11, 3, 7, 9, 5, 6, 2, 8, 1, 9, 11, 10, 8, 12, 6,
+                                  14, 6, 5, 16, 3, 17, 2, 20, 18, 24, 17, 25, 3, 16, 12);
+        return randInts;
+      `
+    }
+  ],
+
+  methods: [
+    {
```

Comment:
Add helper methods 
getNextLong
getNextString
Then we don't have convert to String and back to Long in the use case earlier.  
generate can return type Object, then the helpers can just cast.